internal/provisioner/model/model.go

```diff
@@ -134,6 +136,10 @@ func (c *Contour) EnvoyTolerationsExist() bool {
 	return false
 }
 
+func (c *Contour) WatchAllNamespaces() bool {
+	return c.Spec.WatchNamespaces == nil || slices.Contains(c.Spec.WatchNamespaces, corev1.NamespaceAll)
```

Comment:
NamespacesAll is defined as `""` which I don't personally think we should allow through the CRD validation etc. to get here so I think it might be ok to drop that logic

Reply:
nit: could also check `len(c.Spec.WatchNamespaces) == 0` to guard against an empty list getting through (via tests or otherwise)